internal/worker/server.go

```diff
@@ -115,6 +115,46 @@ func (s *Server) JobStatus(id uuid.UUID) (*JobStatus, error) {
 	}, nil
 }
 
+// Provides access to artifacts of a job. Returns an io.Reader for the artifact
+// and the artifact's size.
+func (s *Server) JobArtifact(id uuid.UUID, name string) (io.Reader, int64, error) {
+	status, err := s.JobStatus(id)
+	if err != nil {
+		return nil, 0, err
+	}
+
+	if status.Finished.IsZero() {
```

Comment:
There is not `FAILED` state :wink: A job is either finished or canceled.

Thanks for mentioning though, because this does not handle the canceled case. A `status.IsFinished()` would be much more readable indeed. I'll fix it up in a bit. Thanks!